task/rpm-ostree/0.1/rpm-ostree.yaml

```diff
@@ -0,0 +1,230 @@
+    name: build
+    resources:
+      limits:
+        cpu: "2"
+        memory: 4402341478400m
+      requests:
+        cpu: 250m
+        memory: 512Mi
+    script: |-
+      set -o verbose
+      set -eu
+      set -o pipefail
+      mkdir -p ~/.ssh
+      if [ -e "/ssh/error" ]; then
+        #no server could be provisioned
+        cat /ssh/error
+        exit 1
+      fi
+      cp /ssh/id_rsa ~/.ssh
+      chmod 0400 ~/.ssh/id_rsa
+      export SSH_HOST=$(cat /ssh/host)
+      export BUILD_DIR=$(cat /ssh/user-dir)
+      export SSH_ARGS="-o StrictHostKeyChecking=no"
+      mkdir -p scripts
+      echo "$BUILD_DIR"
+      ssh $SSH_ARGS "$SSH_HOST"  mkdir -p "$BUILD_DIR/workspaces" "$BUILD_DIR/scripts" "$BUILD_DIR/tmp"
+
+      rsync -ra $(workspaces.source.path)/ "$SSH_HOST:$BUILD_DIR/workspaces/source/"
+      cat >scripts/script-build.sh <<'REMOTESSHEOF'
+      #!/bin/sh
+      set -o verbose
+      cd $(workspaces.source.path)
+      rpm-ostree compose image --initialize-mode always --format oci "source/$IMAGE_FILE" rhtap-final-image
+
+      REMOTESSHEOF
+      chmod +x scripts/script-build.sh
+      rsync -ra scripts "$SSH_HOST:$BUILD_DIR"
+      ssh $SSH_ARGS "$SSH_HOST" podman run --mount type=bind,source=$BUILD_DIR/tmp,target=/var/tmp,relabel=shared --privileged -e CONTEXT="$CONTEXT" -e IMAGE_FILE="$IMAGE_FILE" -e IMAGE="$IMAGE" -e IMAGE_EXPIRES_AFTER="$IMAGE_EXPIRES_AFTER" -e COMMIT_SHA="$COMMIT_SHA"  --rm  -v "$BUILD_DIR/workspaces/source:$(workspaces.source.path):Z"  -v $BUILD_DIR/scripts:/script:Z --user=0  "$BUILDER_IMAGE" /script/script-build.sh
+      rsync -ra "$SSH_HOST:$BUILD_DIR/workspaces/source/" "$(workspaces.source.path)/"
+      cp -r rhtap-final-image /var/lib/containers/rhtap-final-image
+      buildah pull oci:rhtap-final-image
+      buildah images
+      buildah tag localhost/rhtap-final-image "$IMAGE"
+      container=$(buildah from --pull-never "$IMAGE")
+    securityContext:
+      capabilities:
+        add:
+        - SETFCAP
+    volumeMounts:
+    - mountPath: /var/lib/containers
+      name: varlibcontainers
+    - mountPath: /ssh
+      name: ssh
+      readOnly: true
+    workingDir: $(workspaces.source.path)
+  - image: quay.io/redhat-appstudio/syft:v0.94.0
+    name: sbom-syft-generate
+    resources:
+      limits:
+        memory: 6Gi
+      requests:
+        memory: 6Gi
+    script: |
+      syft oci-dir:/var/lib/containers/rhtap-final-image --file=$(workspaces.source.path)/sbom-cyclonedx.json --output=cyclonedx-json
+    volumeMounts:
+      - mountPath: /var/lib/containers
+        name: varlibcontainers
+  - image: registry.access.redhat.com/ubi9/python-39:1-143.1696863474
+    name: create-purl-sbom
+    resources: {}
+    script: |
+      #!/bin/python3
+      import json
+
+      with open("./sbom-cyclonedx.json") as f:
+        cyclonedx_sbom = json.load(f)
+
+      purls = [{"purl": component["purl"]} for component in cyclonedx_sbom["components"] if "purl" in component]
```

Comment:
There was an update to this line for other build tasks: `cyclonedx_sbom.get("components", [])`

Reply:
Thanks, I have made the same change.